src/sql/platform/query/common/gridDataProvider.ts

```diff
@@ -95,6 +95,13 @@ export async function getResultsString(provider: IGridDataProvider, selection: S
 		}
 		await p;
 	}
+	headers = new Map([...headers].sort(([k1], [k2]): number => {
+		return k1 - k2;
+	}));
+
+	rows = new Map([...rows].sort(([k1], [k2]): number => {
```

Comment:
Will be cleaner if we defined a helper function and then take `rows` and `headers` since they are basically using the same code.